src/test/java/org/opentripplanner/raptor/moduletests/K02_TransitPriorityDestinationTest.java

```diff
@@ -0,0 +1,148 @@
+import static org.opentripplanner.raptor._data.RaptorTestConstants.T01_00;
+import static org.opentripplanner.raptor._data.api.PathUtils.pathsToString;
+import static org.opentripplanner.raptor._data.transit.TestAccessEgress.walk;
+import static org.opentripplanner.raptor._data.transit.TestRoute.route;
+import static org.opentripplanner.raptor._data.transit.TestTripPattern.pattern;
+import static org.opentripplanner.raptor._data.transit.TestTripSchedule.schedule;
+
+import java.time.Duration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.raptor.RaptorService;
+import org.opentripplanner.raptor._data.transit.TestTransitData;
+import org.opentripplanner.raptor._data.transit.TestTripSchedule;
+import org.opentripplanner.raptor.api.model.DominanceFunction;
+import org.opentripplanner.raptor.api.request.RaptorProfile;
+import org.opentripplanner.raptor.api.request.RaptorRequestBuilder;
+import org.opentripplanner.raptor.api.request.RaptorTransitPriorityGroupCalculator;
+import org.opentripplanner.raptor.configure.RaptorConfig;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.cost.RaptorCostConverter;
+
+/**
+ * FEATURE UNDER TEST
+ *
+ * Raptor should be able to handle route request with transit-priority.
+ */
+public class K02_TransitPriorityDestinationTest {
+
+  private static final RaptorTransitPriorityGroupCalculator PRIORITY_GROUP_CALCULATOR = new RaptorTransitPriorityGroupCalculator() {
+    @Override
+    public int mergeTransitPriorityGroupIds(int currentGroupIds, int boardingGroupId) {
+      return currentGroupIds | boardingGroupId;
+    }
+
+    /**
+     * Left dominate right, if right has at least one priority group not in left.
+     */
+    @Override
+    public DominanceFunction dominanceFunction() {
+      return (l, r) -> ((l ^ r) & r) != 0;
+    }
+  };
+
+  private static final int GROUP_A = 0x01;
+  private static final int GROUP_B = 0x02;
+  private static final int GROUP_C = 0x04;
+  private static final int GROUP_AB = PRIORITY_GROUP_CALCULATOR.mergeTransitPriorityGroupIds(
+    GROUP_A,
+    GROUP_B
+  );
+  private static final int GROUP_AC = PRIORITY_GROUP_CALCULATOR.mergeTransitPriorityGroupIds(
+    GROUP_A,
+    GROUP_C
+  );
+  private static final int C1_SLACK_90s = RaptorCostConverter.toRaptorCost(90);
+
+  private final TestTransitData data = new TestTransitData();
+  private final RaptorRequestBuilder<TestTripSchedule> requestBuilder = new RaptorRequestBuilder<>();
+  private final RaptorService<TestTripSchedule> raptorService = new RaptorService<>(
+    RaptorConfig.defaultConfigForTest()
+  );
+
+  @BeforeEach
+  private void prepareRequest() {
+    // Each pattern depart at the same time, but arrive with 60s between them.
+    // Given a slack on the cost equals to ~90s make both L1 and L2 optimal, but no L3.
+    // Line U1 is not optimal, because it slower than L1 and is in the same group
```

Comment:
Same comment and suggestion as in `K01_TransitPriorityTest.java`.
```suggestion
    // Each pattern depart at the same time, but arrive at different times and they may
    // belong to different groups.
    // Line U1 is not optimal, because it slower than L1 and is in the same group. 
    // Given a slack on the cost equals to ~90s this makes both L1 and L2 optimal (since
    // they are in different groups), but not L3 (which certainly is in its own group but
    // its cost is outside the range allowed by the slack).
```

Reply:
> which certainly is in its...

I do not like "fluffy" words like _certainly_ in technical documentation, but I will merge this - it is not important enough to me.